# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/WarpCacheIRTranspiler.cpp
# Commit: 954c1651adb2
# Full Hash: 954c1651adb203c44392e127d6487aec3de53c97
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-07-07 09:47:47
# Regressor Bug: 1646378
# File Overlap Count: 1
# Description:
#   Bug 1646378: Add CallInlinedFunction r=jandem
#   
#   This is mostly the same as CallScriptedFunction. The key differences are:
#   
#   1. We have an extra argument, the ICScript, which must be stored in the JSContext so the Baseline prologue can find it.
# ==============================================================================

diff -r 01851025fa25 -r 954c1651adb2 js/src/jit/WarpCacheIRTranspiler.cpp
--- a/js/src/jit/WarpCacheIRTranspiler.cpp	Thu Jul 02 23:02:23 2020 +0000
+++ b/js/src/jit/WarpCacheIRTranspiler.cpp	Mon Jul 06 19:05:21 2020 +0000
@@ -1708,6 +1708,13 @@
   return emitCallFunction(calleeId, argcId, flags, CallKind::Scripted);
 }
 
+bool WarpCacheIRTranspiler::emitCallInlinedFunction(ObjOperandId calleeId,
+                                                    Int32OperandId argcId,
+                                                    uint32_t icScriptOffset,
+                                                    CallFlags flags) {
+  return emitCallFunction(calleeId, argcId, flags, CallKind::Scripted);
+}
+
 // TODO: rename the MetaTwoByte op when IonBuilder is gone.
 bool WarpCacheIRTranspiler::emitMetaTwoByte(MetaTwoByteKind kind,
                                             uint32_t functionObjectOffset,